bluebrain/repo-bluebrain/packages/coreneuron/package.py

```diff
@@ -131,21 +131,27 @@ def profiling_wrapper_off(self):
 
     def get_flags(self):
         spec = self.spec
-        flags = ''
+        flags = []
         if '%intel' in spec:
-            flags = '-xHost -qopt-report=5'
+            flags.append('-qopt-report=5')
             if '+knl' in spec:
-                flags = '-xMIC-AVX512 -qopt-report=5'
+                flags.append('-xMIC-AVX512')
+            else:
+                flags.append('-xHost')
```

Comment:
I don't think this PR makes things any better/worse, but I am not sure how this interacts with Spack, which is probably adding `-march=skylake` in addition to these flags when we build on BB5. It's not immediately clear to me which option "wins".

Reply:
Answering my own question by looking at some `coreneuron%intel` logs on BB5:
```console
[olupton@bbpv2 ~]$ grep march /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_intel-2021.4.0-skylake/coreneuron-1.0.0.20220304-lakpvp/.spack/spack-build-out.txt | sort -u
icc: command line warning #10121: overriding '-march=skylake' with '-xMIC-AVX512'
icpc: command line warning #10121: overriding '-march=skylake' with '-xMIC-AVX512'
[olupton@bbpv2 ~]$ grep march /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_intel-2021.4.0-skylake/coreneuron-1.0.0.20220304-on7dvp/.spack/spack-build-out.txt | sort -u
icc: command line warning #10121: overriding '-march=skylake' with '-xHost'
icpc: command line warning #10121: overriding '-march=skylake' with '-xHost'
```
it seems like this is doing something sane, although in future we might prefer to let Spack steer this instead of doing it manually.